sampleExtension/statisticsPage/statistics.js

```diff
@@ -0,0 +1,131 @@
+
+// Log console messages to the background page console instead of the content page.
+var console = chrome.extension.getBackgroundPage().console;
+
+var bg = chrome.extension.getBackgroundPage();
+
+//Local variables that holds the list of links and interceptCounter.
+var links = [];
+var interceptionCounter = 0;
+var interceptDateList = [];
+var date24 = 0;
+var date7 = 0;
+var date31 = 0;
+
+var html_table = $('#interceptTable');
+var html_intCnt = $('#iCounter');
+var html_date24 = $('#date24');
+var html_date7 = $('#date7');
+var html_date31 = $('#date31');
+
+saveCurrentPageToBlacklist = function() {
+    chrome.tabs.query({active: true, currentWindow: true}, function (arrayOfTabs) {
+        // since only one tab should be active and in the current window at once
+        // the return variable should only have one entry
+        var activeTab = arrayOfTabs[0];
+        var activeTabUrl = activeTab.url;
+        bg.addToBlockedSites(activeTabUrl);
+    });
+}
+
+generateHtmlTableRow = function(site) {
```

Comment:
Good attempt at isolating the generation of table rows, but avoid html in your javascript.